src/tasks/birthday.ts

```diff
@@ -0,0 +1,78 @@
+import { getHandler } from '#languages';
+import { PartialResponseValue, ResponseType, Task } from '#lib/database';
+import { TextChannel, User } from 'discord.js';
+
+const enum Matches {
+	Age = '{age}',
+	AgeOrdinal = '{ageOrdinal}',
+	User = '{user}'
+}
+
+export class UserTask extends Task {
+	private kTransformMessageRegExp = /{age}|{ageOrdinal}|{user}/g;
+
+	public async run(data: BirthdayTaskData): Promise<PartialResponseValue | null> {
+		const guild = this.context.client.guilds.cache.get(data.guildID);
+		if (!guild) return null;
+		const member = await guild.members.fetch(data.userID);
+		if (!member) return null;
+
+		const t = await guild.fetchT();
+		const [birthdayRole, birthdayChannel, birthdayMessage] = await guild.readSettings(['birthdayRole', 'birthdayChannel', 'birthdayMessage']);
+		if (!birthdayRole && !(birthdayChannel && birthdayMessage)) return null;
+		const { me } = guild;
```

Comment:
Never going to be null:
```suggestion
		const me = guild.me!;
```